src/esp/scene/SceneNode.cpp

```diff
@@ -3,35 +3,191 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "SceneNode.h"
+#include "SceneGraph.h"
+#include "esp/core/Check.h"
 #include "esp/geo/geo.h"
+#include "esp/sensor/Sensor.h"
 
 namespace Mn = Magnum;
 
 namespace esp {
 namespace scene {
 
-SceneNode::SceneNode(SceneNode& parent)
-    : Mn::SceneGraph::AbstractFeature3D{*this} {
-  setParent(&parent);
-  setId(parent.getId());
+SceneNode::SceneNode() : Mn::SceneGraph::AbstractFeature3D{*this} {
   setCachedTransformations(Mn::SceneGraph::CachedTransformation::Absolute);
   absoluteTransformation_ = absoluteTransformation();
+  // Once created, nodeSensorSuite_ and subtreeSensorSuite_ are features owned
+  // by the SceneNode. No need to release them in the destructor since the
+  // magnum SceneGraph will handle it.
+  nodeSensorSuite_ = new esp::sensor::SensorSuite(*this);
+  subtreeSensorSuite_ = new esp::sensor::SensorSuite(*this);
 }
 
-SceneNode::SceneNode(MagnumScene& parentNode)
-    : Mn::SceneGraph::AbstractFeature3D{*this} {
-  setParent(&parentNode);
-  setCachedTransformations(Mn::SceneGraph::CachedTransformation::Absolute);
-  absoluteTransformation_ = absoluteTransformation();
+SceneNode::SceneNode(SceneNode& parent) : SceneNode() {
```

Comment:
If `SceneNode()` constructor is removed, can this one call the delegating constructor `SceneNode::SceneNode(MagnumScene& parentNode)` below?
 You may need a type conversion in the argument.